site/mkdocs.yml

```diff
@@ -22,11 +22,56 @@ extra_css:
+        spark_jar_package: org.apache.iceberg:iceberg-spark-runtime-3.3_2.13:0.14.1
+    nessie_spark_extensions_33_213:
+        all_versions_url: https://search.maven.org/artifact/org.projectnessie/nessie-spark-extensions-3.3_2.13
+        jar_url: https://repo.maven.apache.org/maven2/org/projectnessie/nessie-spark-extensions-3.3_2.13/0.43.0/nessie-spark-extensions-3.3_2.13-0.43.0.jar
+        spark_jar_package: org.projectnessie:nessie-spark-extensions-3.3_2.13:0.43.0
+    # Spark 3.2 and Scala 2.12:
+    iceberg_spark_runtime_32_212:
+        all_versions_url: https://search.maven.org/artifact/org.apache.iceberg/iceberg-spark-runtime-3.2_2.12
+        jar_url: https://repo.maven.apache.org/maven2/org/apache/iceberg/iceberg-spark-runtime-3.2_2.12/0.14.1/iceberg-spark-runtime-3.2_2.12-0.14.1.jar
+        spark_jar_package: org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:0.14.1
+    nessie_spark_extensions_32_212:
+        all_versions_url: https://search.maven.org/artifact/org.projectnessie/nessie-spark-extensions-3.2_2.12
+        jar_url: https://repo.maven.apache.org/maven2/org/projectnessie/nessie-spark-extensions-3.2_2.12/0.43.0/nessie-spark-extensions-3.2_2.12-0.43.0.jar
+        spark_jar_package: org.projectnessie:nessie-spark-extensions-3.2_2.12:0.43.0
+    # Spark 3.2 and Scala 2.13:
+    iceberg_spark_runtime_32_213:
+        all_versions_url: https://search.maven.org/artifact/org.apache.iceberg/iceberg-spark-runtime-3.2_2.13
+        jar_url: https://repo.maven.apache.org/maven2/org/apache/iceberg/iceberg-spark-runtime-3.2_2.13/0.14.1/iceberg-spark-runtime-3.2_2.13-0.14.1.jar
+        spark_jar_package: org.apache.iceberg:iceberg-spark-runtime-3.2_2.13:0.14.1
+    nessie_spark_extensions_32_213:
+        all_versions_url: https://search.maven.org/artifact/org.projectnessie/nessie-spark-extensions-3.2_2.13
+        jar_url: https://repo.maven.apache.org/maven2/org/projectnessie/nessie-spark-extensions-3.2_2.13/0.43.0/nessie-spark-extensions-3.2_2.13-0.43.0.jar
+        spark_jar_package: org.projectnessie:nessie-spark-extensions-3.2_2.13:0.43.0    
+    # Spark 3.1 and Scala 2.12:
+    iceberg_spark_runtime_31_212:
+        all_versions_url: https://search.maven.org/artifact/org.apache.iceberg/iceberg-spark-runtime-3.1_2.12
+        jar_url: https://repo.maven.apache.org/maven2/org/apache/iceberg/iceberg-spark-runtime-3.1_2.12/0.14.1/iceberg-spark-runtime-3.1_2.12-0.14.1.jar
+        spark_jar_package: org.apache.iceberg:iceberg-spark-runtime-3.1_2.12:0.14.1
+    nessie_spark_extensions_31_212:
+        all_versions_url: https://search.maven.org/artifact/org.projectnessie/nessie-spark-extensions-3.1_2.12
+        jar_url: https://repo.maven.apache.org/maven2/org/projectnessie/nessie-spark-extensions-3.1_2.12/0.43.0/nessie-spark-extensions-3.1_2.12-0.43.0.jar
+        spark_jar_package: org.projectnessie:nessie-spark-extensions-3.1_2.12:0.43.0
```

Comment:
This the aforementioned "data structure" that holds all the identifiers/urls for the artifacts.

It's ugly, and there's duplication.

But the good side of this is that all the duplication is here in one place, the URLs are well-formed _(and easily validatable/visitable/clickable)_, there's no more arbitrary `"{{spark_33}}_{{ scala_212 }}"`-style jinja-junk copy-pasted over the documentation.

Reply:
Hm. The structure would have to be manually adjusted for every Iceberg and every Nessie release. Iceberg 1.0.0 has been released yesterday and Nessie's getting released today.
The pattern looks like it can be generated. For Spark 3.1, there's only Scala 2.12, for Spark 3.2+3.3 there's 2.12+2.13.

Reply:
I agree, it does look like it can be generated - but only from `0.14.0` forward.  Nessie and Iceberg are in alignment with their artifact names from that point forward, but if we want to provide support for old artifact versions, they won't follow the visible pattern.  This style can support those older releases.

But, maintenance is key - or rather the lack of a need for it - so I'll experiment a little bit more and see if I can adjust the pattern to require no manual adjustment.

Reply:
Thanks for pushing me farther here @snazy.  I experimented a bit - and I've generated the pattern with no need for any boilerplate.  Please give [the latest](https://github.com/projectnessie/nessie/pull/5371#pullrequestreview-1148261986) a review when you can find the time.